backend/app/deployment_handler.go

```diff
@@ -722,3 +775,248 @@ func (h *Handler) HandleRemoveNode(c *gin.Context) {
+		vm.Node.EnvVars = map[string]string{}
+	}
+
+	if vm.Network.Name == "" {
+		vm.Network.Name = vm.ProjectName + "net"
+	}
+
+	if err := internal.ValidateStruct(vm.Node); err != nil {
+		Error(c, http.StatusBadRequest, "Validation failed", err.Error())
+		return
+	}
+
+	wf, err := h.ewfEngine.NewWorkflow(activities.WorkflowDeployVM)
+	if err != nil {
+		InternalServerError(c)
+		return
+	}
+
+	wf.State = ewf.State{
+		"config": config,
+		"vm":     vm,
+	}
+
+	h.ewfEngine.RunAsync(c, wf)
+
+	Success(c, http.StatusAccepted, "VM deployment in progress", DeployVMResponse{
+		WorkflowID: wf.UUID,
+		Status:     string(wf.Status),
+	})
+
+}
+
+// @Summary List VMs
+// @Description Get all virtual machines for the authenticated user
+// @Tags vms
+// @Security BearerAuth
+// @Accept json
+// @Produce json
+// @Success 200 {object} ListVMsResponse "Vms are listed successfully"
+// @Failure 401 {object} APIResponse "Unauthorized"
+// @Failure 500 {object} APIResponse "Internal server error"
+// @Router /deployments/vms [get]
+func (h *Handler) HandleListVMs(c *gin.Context) {
+	userID, exists := c.Get("user_id")
+	if !exists {
+		Error(c, http.StatusUnauthorized, "Unauthorized", "user not authenticated")
+		return
+	}
+
+	id := fmt.Sprintf("%v", userID)
+	vms, err := h.db.ListUserVMS(id)
+	if err != nil {
+		logger.GetLogger().Error().Err(err).Str("user_id", id).Msg("Failed to list user VMs")
+		InternalServerError(c)
+		return
+	}
+
+	vmList := make([]VMItem, 0, len(vms))
+	for _, vm := range vms {
+		vmResult, err := vm.GetVMResult()
+		if err != nil {
+			logger.GetLogger().Error().Err(err).Int("vm_id", vm.ID).Msg("Failed to deserialize VM result")
+			continue
+		}
+
+		vmList = append(vmList, VMItem{
+			ID:          vm.ID,
+			ProjectName: vm.ProjectName,
+			VM:          vmResult,
+			CreatedAt:   vm.CreatedAt,
+		})
+	}
+
+	Success(c, http.StatusOK, "Vms are listed successfully", ListVMsResponse{
+		VMs:   vmList,
+		Count: len(vmList),
+	})
+}
+
+// @Summary List VM
+// @Description Get specific VM for authenticated user
+// @Tags vms
+// @Security BearerAuth
+// @Accept json
+// @Produce json
+// @Param id path string true "VM ID"
+// @Success 200 {object} ListVMResponse
+// @Failure 400 {object} APIResponse "Bad Request"
+// @Failure 401 {object} APIResponse "Unauthorized"
+// @Failure 404 {object} APIResponse "VM not found"
+// @Failure 500 {object} APIResponse "Internal server error"
+// @Router /deployments/vms/{id} [get]
+func (h *Handler) HandleListVM(c *gin.Context) {
+	config, err := h.getClientConfig(c)
+	if err != nil {
+		InternalServerError(c)
+		return
+	}
+
+	vmID := c.Param("id")
+	if vmID == "" {
+		Error(c, http.StatusBadRequest, "Vm id is required", "")
+		return
+	}
+
+	vm, err := h.db.GetVMByID(config.UserID, vmID)
+	if err != nil {
+		logger.GetLogger().Error().Err(err).
+			Str("user_id", config.UserID).
+			Str("vm_id", vmID).
+			Msg("Failed to find VM")
+		Error(c, http.StatusNotFound, "Vm is not found", err.Error())
```

Comment:
Check on the error if it's of type not found or not. Just don't send the internal database error to the user.
If any other type of errors, return InternalServerError